Init BindGen README (#8764)
ethereum-optimism/optimism: op-bindings/bindgen/README.md

5. Compares the deployment bytecode retrieved from Etherscan on Ethereum Mainnet against the bytecode retrieved from Etherscan on Optimism Mainnet
  - This has the same concern as differing initialization bytecode
6. Lastly, the Go bindings are generated and the metadata file is written to the path provided as `metadata-out` CLI flag

All other default `"remote"` contract have some variation of the above execution flow depending on the nuances of each contract. For example:

- `Create2Deployer`'s initialization and deployed bytecode is expected to differ from it's Optimism Mainnet deployment
- `MultiSend_v130` has an `immutable` Solidity variable the resolves to `address(this)`, so we can't use the deployment bytecode from Ethereum Mainnet, we must get it's deployment bytecode from Optimism Mainnet
- `SenderCreator` is deployed by `EntryPoint`, so it's initialization bytecode is provided in [artifacts.json](../artifacts.json) and not being fetched from Etherscan like other contracts

#### Contracts that Don't Make Good Predeploys

> Not every contract can be added as a predeploy, and some contract's have nuances that make them potentially dangerous or troublesome to predeploy. Some examples of contracts that wouldn't make good predeploys:

Reviewer: ```suggestion
Not every contract can be added as a predeploy, and some contracts have nuances that make them potentially dangerous or troublesome to predeploy. Below are some examples of contracts that wouldn't make good predeploys. This is not a comprehensive list, so make sure to use judgment for each contract added as a predeploy.
```

- Contracts that haven't been audited or stood the test of time

>   - Once a contract is predeployed and a network is started, if a vulnerability is discovered for the contract and there is no way to disable the contract using some variable, the only option to "disable" the vulnerable contract is to either A. remove it from the L2 genesis and restart the L2 network B. Hardfork the network to remove/replace the predeploy or C. Warn users not to use the vulnerable predeploy

Reviewer: nit to make the list more readable
```suggestion
  - Once a contract is predeployed and a network is started, if a vulnerability is discovered for the contract and there is no way to easily disable the contract, the only options to "disable" the vulnerable contract are to either (A) remove it from the L2 genesis and restart the L2 network, (B) Hardfork the network to remove/replace the predeploy, or (C) Warn users not to use the vulnerable predeploy
```

- Related to above, contracts that may become deprecated/unsupported relatively soon
  - As mentioned above, you're limited to options A, B, or C

> - Upgradeable Contracts
>   - While it's certainly feasible to predeploy an upgradeable contract, great care should be taken to ensure user's aren't "rugged" if the contract is upgraded to a malicious implementation. Understanding who has the ability to upgrade the contract is key to avoiding this. Additionally, user's might be expecting a predeploy to do something and may be caught off guard if the implementation was upgraded without their knowledge

Reviewer: Replaced "rugged" with "security risks" wording, and added a new contract type
```suggestion
- Upgradeable Contracts
  - While it's certainly feasible to predeploy an upgradeable contract, great care should be taken to minimize security risks to users if the contract is upgraded to a malicious or buggy implementation. Understanding who has the ability to upgrade the contract is key to avoiding this. Additionally, user's might be expecting a predeploy to do something and may be caught off guard if the implementation was upgraded without their knowledge
- Contracts with Privileged Roles and Configuration Parameters
  - Similar to the upgradeable contracts, simply having an owner or other privileged role with the ability to make configuration changes can present a security risk and result in unexpected different behaviors across chains.
```

Flag               | Type   | Description                                                   | Required
------------------ | ------ | ------------------------------------------------------------- | --------
`source-maps-list` | String | Comma-separated list of contracts to generate source-maps for | No
`forge-artifacts`  | String | Path to the directory with compiled Forge artifacts           | Yes

## Remote Flags

These flags are used with `all` and `remote` commands

Flag                   | Type   | Description                                                                 | Required
---------------------- | ------ | --------------------------------------------------------------------------- | --------
`etherscan.apikey.eth` | String | An Etherscan API key for querying Ethereum Mainnet                          | Yes
`etherscan.apikey.op`  | String | An Etherscan API key for querying Optimism Mainnet                          | Yes
`rpc.url.eth`          | String | This is any HTTP URL that can be used to query an Ethereum Mainnet RPC node | Yes
`rpc.url.op`           | String | This is any HTTP URL that can be used to query an Optimism Mainnet RPC node | Yes

> # Using BindGen to Add New Predeploys to L2 Genesis

Reviewer 2: Can we add a small snippet here to specify that we're not looking to endorse more contracts at this moment? Something like 

> While we encourage hacking on the stack, we are not actively looking to integrate more contracts at an OP Stack's genesis.

Reviewer 3: I added

```markdown
**Note** While we encourage hacking on the OP stack, we are not actively looking to integrate more contracts at the SuperChain's genesis.
```

Author: Updated to

```markdown
**Note** While we encourage hacking on the OP stack, we are not actively looking to integrate more contracts to the official OP stack genesis.
```

5. Compares the deployment bytecode retrieved from Etherscan on Ethereum Mainnet against the bytecode retrieved from Etherscan on Optimism Mainnet
  - This has the same concern as differing initialization bytecode
6. Lastly, the Go bindings are generated and the metadata file is written to the path provided as `metadata-out` CLI flag

All other default `"remote"` contract have some variation of the above execution flow depending on the nuances of each contract. For example:

- `Create2Deployer`'s initialization and deployed bytecode is expected to differ from it's Optimism Mainnet deployment
- `MultiSend_v130` has an `immutable` Solidity variable the resolves to `address(this)`, so we can't use the deployment bytecode from Ethereum Mainnet, we must get it's deployment bytecode from Optimism Mainnet
- `SenderCreator` is deployed by `EntryPoint`, so it's initialization bytecode is provided in [artifacts.json](../artifacts.json) and not being fetched from Etherscan like other contracts

> #### Contracts that Don't Make Good Predeploys

Reviewer 2: this section is great!
